src/EnduranceTests/Handlers/HeartbeatHandler.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace NServiceBus.AzureServiceBus.EnduranceTests.Handlers
+{
+    using System.Threading.Tasks;
+    using Commands;
+    using Logging;
+
+    public class HeartbeatHandler : IHandleMessages<Heartbeat>
+    {
+        static readonly ILog log = LogManager.GetLogger<HeartbeatHandler>();
+
+        public async Task Handle(Heartbeat message, IMessageHandlerContext context)
+        {
+            log.InfoFormat("HeartbeatHandler Heartbeat with Wait of {0}", TestSettings.Rate);
+
+            if (TestSettings.TestRunId != message.TestRunId)
```

Comment:
This removes any lingering messages from earlier test runs. Alternative to trying to clear the queue before starting the test.